CRM/Contribute/BAO/Contribution.php

```diff
@@ -5457,4 +5457,32 @@ public static function createProportionalFinancialEntries($entityParams, $lineIt
     }
   }
 
+  /**
+   * Calculate Tax when Financial Type is changed.
+   *
+   * @param array $params
+   * @param float $totalAmount
+   * @param array $oldTaxAmounts
```

Comment:
Please describe in here what values are expected to exist in $oldTaxAmounts. Without a description, I would think this would be a list a numbers, but looking at the code I see it's not